Exercise10.R

```diff
@@ -0,0 +1,65 @@
+#set working directory and load ggplot
+setwd("/users/sydneyharris/Desktop/")
+library(ggplot2)
+
+#Question 1
+#read data into R and save as variable
+UW_MSU <- read.table("/users/sydneyharris/desktop/UWvMSU_1-22-13.txt", header = T, sep = "\t")
+
+#pre-allocate vectors to hold information
+UW_scores = numeric(length(c(1:nrow(UW_MSU)))) #points gained by UW every time a basket is scored regardless of team
+MSU_scores = numeric(length(c(1:nrow(UW_MSU)))) #points gained by MSU every time a basket is scores regardless of team
+
+#for loop with if-else statements
+for (i in 1:nrow(UW_MSU)){
+  if (UW_MSU$team[i] == "UW"){
+#for each row in the data set, if the team is UW, it will store the points gained in the vector UW_scores
+#it will store a 0 in the vector MSU_scores because MSU gained 0 points
+    UW_scores[i] <- UW_MSU$score[i]
+    MSU_scores[i] <- 0
+  }else if (UW_MSU$team[i] == "MSU"){
+#for each row in the data set, if the team is MSU, it will store the points gained in the vector MSU_scores
+#it will store a 0 in the vector UW_scores because UW gained 0 points
+    MSU_scores[i] <- UW_MSU$score[i]
+    UW_scores[i] <- 0
+  }}
+
+#creates 2 columns in UW_MSU called UW_Total and MSU_Total with the cumulative sum of each team when a basket is scored regardless of team
+UW_MSU$UW_Total <- c(cumsum(UW_scores))
+UW_MSU$MSU_Total <- c(cumsum(MSU_scores))
```

Comment:
You can use the for loops for cumulative sum as well. Checking this code:

# load table of scoring
scoring=read.table("UWvMSU_1-22-13.txt",header=TRUE,sep="\t",stringsAsFactors=FALSE)

# look at data
dim(scoring)
head(scoring)

# preallocating matrix to store cumulative scores
cum_scores=matrix(NA,nrow(scoring)+1,3)
cum_scores[,1]=c(0,scoring[,1])
cum_scores[1,2:3]=0
colnames(cum_scores)=c("time","UW","MSU")

# looping through individual scoring events
for(i in 1:nrow(scoring)){
  if(scoring[i,2]=="UW"){
    cum_scores[(i+1),2]=cum_scores[i,2]+scoring[i,3]
    cum_scores[(i+1),3]=cum_scores[i,3]
  }else{
    cum_scores[(i+1),2]=cum_scores[i,2]
    cum_scores[(i+1),3]=cum_scores[i,3]+scoring[i,3]
  }
}